src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -0,0 +1,412 @@
+/*
+ * citus_depended_object.c
+ *
+ * Implements the functions for hiding citus objects.
+ *
+ * Copyright (c) Citus Data, Inc.
+ */
+
+#include "postgres.h"
+#include "miscadmin.h"
+
+#include "access/genam.h"
+#include "catalog/pg_aggregate.h"
+#include "catalog/pg_am.h"
+#include "catalog/pg_attribute.h"
+#include "catalog/pg_attrdef.h"
+#include "catalog/pg_constraint.h"
+#include "catalog/pg_class.h"
+#include "catalog/pg_depend.h"
+#include "catalog/pg_enum.h"
+#include "catalog/pg_event_trigger.h"
+#include "catalog/pg_extension.h"
+#include "catalog/pg_language.h"
+#include "catalog/pg_namespace.h"
+#include "catalog/pg_proc.h"
+#include "catalog/pg_rewrite.h"
+#include "catalog/pg_sequence.h"
+#include "catalog/pg_statistic.h"
+#include "catalog/pg_trigger.h"
+#include "catalog/pg_ts_config.h"
+#include "catalog/pg_ts_dict.h"
+#include "catalog/pg_ts_template.h"
+#include "catalog/pg_type.h"
+#include "catalog/namespace.h"
+#include "commands/defrem.h"
+#include "commands/extension.h"
+#include "common/hashfn.h"
+#include "distributed/citus_depended_object.h"
+#include "distributed/commands.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/listutils.h"
+#include "distributed/log_utils.h"
+#include "distributed/metadata/dependency.h"
+#include "distributed/metadata/distobject.h"
+#include "nodes/makefuncs.h"
+#include "nodes/nodeFuncs.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+#include "utils/lsyscache.h"
+#include "utils/syscache.h"
+
+/*
+ * GUC hides any objects, which depends on citus extension, from pg meta class queries,
+ * it is intended to be used in vanilla tests to not break postgres test logs
+ */
+bool HideCitusDependentObjects = false;
+
+static Node * CreateCitusDependentObjectExpr(int pgMetaTableVarno, int pgMetaTableOid);
+static List * GetFuncArgs(int pgMetaTableVarno, int pgMetaTableOid);
+
+PG_FUNCTION_INFO_V1(is_citus_depended_object);
```

Comment:
for the UDFs that we don't want to expose to the users, but only use on the regression tests, there is a folder https://github.com/citusdata/citus/tree/main/src/backend/distributed/test

Can we move `src/backend/distributed/utils/citus_depended_object.c` under that directory such that this UDF is not exposed other than tests?

(please try if this works with vanilla testing as well, I'm not sure. But, if we can do that, it'd be great. Lets move as much as code under test folder)